### PR TITLE
feat: Script Injector

### PR DESCRIPTION
**Pull request by BlankParticle**
This PR introduces a Script Injector, updates dependencies and add Nix Flakes dev env support.
JavaScript can be injected in `src/injector.js` file

### CODE REVIEW

File: src-tauri/tauri.conf.json
Comment (ghost-2e63) [inline src-tauri/tauri.conf.json:1]: 'dangerousRemoteDomainIpcAccess' is not allowed
  Additional property 'userAgent' is not allowed

File: src/injection.js [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+      window.localStorage.removeItem("dark");
+      styleEl.remove();
+    } else {
+      window.localStorage.setItem("dark", "true");
+      document.head.append(styleEl);
+    }
+  };
+
+  window.addEventListener("keypress", (key) => {
+    if (key.key == "i" && key.ctrlKey) {
```
Comment (ghost-2e63) [inline src/injection.js:97]: Keybinds are not working on windows.

File: src-tauri/tauri.conf.json
Comment (BlankParticle) [reply src-tauri/tauri.conf.json:1]: I think you didn't update the dependencies, I just cloned the repo on my windows machine and tried, It worked fine. Also the keybinds need a fix for windows, commiting that now

**Comment by ghost-2e63** [comment]
image

Issue while building for windows.
``tauri build --bundles msi``

**Comment by BlankParticle** [comment]
add "devtools" in cargo.toml
like this
```toml
tauri = {..., features = [...,"devtools"]};
```

File: src-tauri/Cargo.toml [hunk 2]
```diff
@@ -1,35 +1,27 @@
...
-# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
 [build-dependencies]
-tauri-build = { version = "1.1", features = [] }
+tauri-build = { version = "1.4.0", features = [] }
 
 [dependencies]
-serde_json = "1.0"
-serde = { version = "1.0", features = ["derive"] }
-tauri = { version = "1.1", features = ["process-relaunch"] }
```
Comment (BlankParticle) [inline src-tauri/Cargo.toml:19]:
```toml
tauri = { version = "1.1", features = ["process-relaunch", "devtools"] }
```

**Pull request merged**